code/lib/instrumenter/src/instrumenter.ts

```diff
@@ -44,8 +44,8 @@ const isObject = (o: unknown) => Object.prototype.toString.call(o) === '[object
 const isModule = (o: unknown) => Object.prototype.toString.call(o) === '[object Module]';
 const isInstrumentable = (o: unknown) => {
   if (!isObject(o) && !isModule(o)) return false;
-  if (o.constructor === undefined) return true;
-  const proto = o.constructor.prototype;
+  if ((o as object).constructor === undefined) return true;
+  const proto = (o as object).constructor.prototype;
```

Comment:
The `isInstrumentable` function's argument is _deliberately_ typed as unknown, because it needs to be treated as such. Casting that argument to `object` defeats the purpose of that, particularly because `object` is basically an alias for `any`. If the goal of enabling strict types is to improve the overall type-security, then this type-casting contradicts that goal. The best way to improve type-security in this instance is to improve the type _hygiene_, rather than enabling `strict: true` for the sake of it.

There are a few different courses of action here. For example, you could _handle_ the case where `o` is nullish, presumably by returning `false` (before doing the `isObject` and `isModule` checks). And/or, you could make the return types of `isObject` and `isModule` [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates).

Reply:
Great point @ryami333 !